models/sla_policy/reply_time/int_zendesk__reply_time_combined.sql

```diff
@@ -137,18 +139,21 @@ with reply_time_calendar_hours_sla as (
                 or (agent_reply_at is null and next_solved_at is null)
       then true
       else false
-        end as is_sla_breached
+        end as is_sla_breached,
+    sum_lapsed_business_minutes_new + total_runtime_minutes as total_new_minutes -- add total runtime to sum_lapsed_business_minutes_new (the sum_lapsed_business_minutes from prior row)
   from filtered_reply_times
-  
-), reply_time_breach as (
-  select 
+
+), reply_time_breach as ( 
+  select  
     *,
-    case when {{ dbt.datediff("sla_schedule_start_at", "agent_reply_at", 'minute') }} < 0 
-      then 0 
-      else sum_lapsed_business_minutes_new + {{ dbt.datediff("sla_schedule_start_at", "coalesce(agent_reply_at, current_time_check)", 'minute') }} 
+    case when total_runtime_minutes < 0 -- agent has already replied to prior to this SLA schedule
+        then 0 -- so don't add new minutes to the SLA
+      when total_new_minutes > sum_lapsed_business_minutes -- if total runtime, regardless of when the SLA schedule ended, is more than the total lapsed business minutes, that means the agent replied after the SLA schedule
+          then sum_lapsed_business_minutes -- the elapsed time after the SLA end time should not be calculated as part of the business minutes, therefore sla_elapsed_time should only be sum_lapsed_business_minutes
+      else total_new_minutes -- otherwise, the sla_elapsed_time will be the prior row's sum_lapsed_business_minutes plus the minutes between SLA schedule start and agent_reply_time.
```

Comment:
I didn't realize that the `total_new_minutes` calculations in this PR is actually difference from the `else` statement in the previous version of the package. We were not accounting for the coalesce (`coalesce(agent_reply_at, current_time_check)`). However, we should not be using the Coalesce in the calculation earlier. 

To be safe I think we should keep the previous calculation to ensure the data is not changing for tickets that were working previously. What are your thoughts?
```suggestion
      else sum_lapsed_business_minutes_new + {{ dbt.datediff("sla_schedule_start_at", "coalesce(agent_reply_at, current_time_check)", 'minute') }} -- otherwise, the sla_elapsed_time will be the prior row's sum_lapsed_business_minutes plus the minutes between SLA schedule start and agent_reply_time.
```

Reply:
Ah good job noticing.  Now that I compare the prior version to current, `total_new_minutes` is not equal to that prior else statement. It makes sense to keep the prior and not risk affecting other tickets.